### PR TITLE
improved structure, added makefile, added checks

### PR DESCRIPTION
**Pull request by spacey-sooty**

### CODE REVIEW

**Comment by spacey-sooty** [comment]
@Vyppal I added some comments please read and take into consideration

File: .github/workflows/checks.yml
Comment (Vyppal) [inline .github/workflows/checks.yml:1]: what?

File: .gitignore
Comment (Vyppal) [inline .gitignore:1]: this was going to be my working branch for my engine, i wanted it to be hidden, hence why i hid hpp and cpp files

Comment (spacey-sooty) [reply .gitignore:1]: ok that makes sense are you still going to do that? I saw your other branch didn't gitignore them. Any I will definitely gitignore things like .exe .out or .gch

File: .github/workflows/checks.yml
Comment (spacey-sooty) [reply .github/workflows/checks.yml:1]: this allows you to just look for the little tick to see if someones code works I can explain in more detail if you need

**Comment by spacey-sooty** [comment]
@Vyppal completed work you requested plus a little bit more

**Pull request merged**